assets/src/edit-story/app/story/useStoryReducer/reducers/restore.js

```diff
@@ -38,7 +38,7 @@ function restore(state, { pages, current, selection, story, capabilities }) {
   const newCurrent = pages.some(({ id }) => id === current)
     ? current
     : pages[0].id;
-  const newSelection = Array.isArray(selection) ? [...new Set(selection)] : [];
+  const newSelection = Array.isArray(selection) ? selection : [];
```

Comment:
`restore` is basically only an internal API used when history is replayed or a story is reloaded from a saved state. There's no need to ensure the array is unique at these points, because the array must have been created by other methods in the API previously, which all makes sure the array was unique at that time and thus still is.

I think it's fair to remove this requirement from this function (and the corresponding test).

Another option here would be to check if the array was unique, and if not, only then re-create the array with a unique'd list, but I don't think it'll ever happen.

Reply:
The bug here was, that the history manager expected the exact same variables (by reference) to be re-entered as a history state when replaying an old state. The old line changed the selection array by reference (though not by content). Even for empty arrays!